src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/CommentHelper.java

```diff
@@ -368,8 +368,12 @@ protected String defaultAction(DocTree node, Void p) {
     }
 
     public TypeElement getReferencedClass(DocTree dtree) {
-        Utils utils = configuration.utils;
         Element e = getReferencedElement(dtree);
+        return getReferencedClass(e);
+    }
+
+    public TypeElement getReferencedClass(Element e) {
+        Utils utils = configuration.utils;
```

Comment:
Whinge. This new method does not really belong here, also implying that the body of the old method does not belong here (in the class) either.  Generally, queries on elements belong in `Utils`, not `CommentHelper`.

But, OK for now, I guess.

--

As a one-off, it was OK to let this slide. Given there are repeated similar occurrences later on, can you file an RFE to move the `Element` / non-`DocTree` methods to `Utils`.